provisioner/clusterpy.go

```diff
@@ -97,6 +97,36 @@ func (p *clusterpyProvisioner) Supports(cluster *api.Cluster) bool {
 	return cluster.Provider == providerID
 }
 
+func (p *clusterpyProvisioner) updateDefaults(cluster *api.Cluster, channelConfig *channel.Config) error {
+	defaultsFile := path.Join(channelConfig.Path, "cluster", "config-defaults.yaml")
+
+	withoutConfigItems := *cluster
+	withoutConfigItems.ConfigItems = make(map[string]string)
+
+	result, err := applyTemplate(newApplyContext(channelConfig.Path), defaultsFile, &withoutConfigItems)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		}
+		return err
+	}
+
+	var defaults map[string]string
+	err = yaml.Unmarshal([]byte(result), &defaults)
+	if err != nil {
+		return err
+	}
+
+	for k, v := range defaults {
+		_, exists := cluster.ConfigItems[k]
+		if !exists {
```

Comment:
nit: it's common in go to use `ok` for this variable, just reads easier if it's the same name used. :)